docs/source/subscriptions.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Adding Subscriptions To Schema
+---
+
+Adding GraphQL subscriptions to your graphQL schema is simple. since Subscriptions are enabled by default in Apollo 2.0 and you dont' need to install anymore any extra packages.
+As example we want to display to display a list of post that contains author and comment (Query) and we want to add a post to them (Mutation).
+
+Our schema will be:
+
+```
+    type Query {
+        posts: [Post]
+    }
+    type Mutation {
+        addPost(author: String, comment: String): Post
+    }
+    type Post {
+        author: String
+        comment: String
+    }
+```
+
+Our resolver:
+```js
+
+const postResolver = {
+    Query: {
+        posts(root: any, args: any, context: any) {
+            return postController.posts();
+        }
+    },
+    Mutation: {
+        addPost(root: any, args: any, context: any) {
+            pubsub.publish(POST_ADDED, { postAdded: args });
+            return postController.addPost(args);
+        }
+    }
+};
+
+export default postResolver;
+
+```
+
+To enable subscription we need to apply just few changes:
+
+- In our schema:
+
+```
+    type Query {
+        posts: [Post]
+    }
+    type Mutation {
+        addPost(author: String, comment: String): Post
+    }
+    type Post {
+        author: String
+        comment: String
+    }
+    type Subscription {
+        postAdded: Post
+    }
+```
+
+We just added `type Subscription`,
+
+- In the resolver:
+```js
+const POST_ADDED = 'POST_ADDED';
+
+const postResolver = {
+    Subscription: {
+        postAdded: {
+            subscribe: () => pubsub.asyncIterator([POST_ADDED])
+        }
+    },
+    Query: {
+		posts(root: any, args: any, context: any) {
+			return postController.posts();
+		}
+    },
+    Mutation: {
+        addPost(root: any, args: any, context: any) {
+            pubsub.publish(POST_ADDED, { postAdded: args });
+            return postController.addPost(args);
+        }
+    }
+};
+
+export default postResolver;
+```
+1. we added in the mutation `pubsub.publish`
+2. we added the Subscription `postAdded`
+
+In our server we don't need to make any big changes:
+1. we need to export
+```js
+export const pubsub = new PubSub();
+```
+2. In case you defined a context remember to add an extra check related to `req.headers` because context will pass the connection in the form of:
+
+```js
+context: ({ connection }) => { ... }`
+```
+A good workaround to this problem is to return in case `req` or `req.headers` are not defined:
```

Comment:
I'm not sure this is a problem, if we think it is, then we should change it. I would suggest that those using subscriptions include connection and req in their context creation function and have checks depending on the input. This is especially important, since the auth tokens are handled differently depending on the transport.

Reply:
I refactor a bit see if this is bette. On the other side without the if statement if you have check on token etc it doesn't work as expected